assertsprocessor/metrics.go

```diff
@@ -141,10 +143,24 @@ func (p *metricHelper) captureMetrics(namespace string, service string, span *pt
 	attrValue, _ := span.Attributes().Get(AssertsRequestContextAttribute)
 	requestContext := attrValue.AsString()
 
-	cache, _ := p.requestContextsByService.LoadOrCompute(serviceKey, func() *ttlcache.Cache[string, string] {
-		cache := ttlcache.New[string, string](
-			ttlcache.WithTTL[string, string](time.Minute*time.Duration(p.config.RequestContextCacheTTL)),
-			ttlcache.WithCapacity[string, string](uint64(p.config.LimitPerService)),
+	cache, _ := p.requestContextsByService.LoadOrCompute(serviceKey, func() *ttlcache.Cache[string, prometheus.Labels] {
+		cache := ttlcache.New[string, prometheus.Labels](
+			ttlcache.WithTTL[string, prometheus.Labels](p.ttl),
+			ttlcache.WithCapacity[string, prometheus.Labels](uint64(p.config.LimitPerService)),
+		)
+		cache.OnEviction(
+			func(ctx context.Context, reason ttlcache.EvictionReason, item *ttlcache.Item[string, prometheus.Labels]) {
+				p.logger.Info("Evicted request context from cache",
+					zap.String("service", serviceKey),
+					zap.String("request context", item.Key()),
+				)
+
+				p.latencyHistogram.Delete(item.Value())
```

Comment:
I think we should consider the `DeletePartialMatch` method instead.

Reply:
Good point, I was thinking an exact match would suffice in this case. But for the reasons you mentioned partial match is a better fit in this case